### PR TITLE
use is_integer in df.insert

### LINKED ISSUE #53193: BUG: df.insert can't accept int64 value
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [x] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
import pandas as pd

df = pd.DataFrame({"a":[1,2]})
loc_a = df.columns.get_indexer(["a"])[0]
df.insert(loc_a, "b", 0)
```

### Issue Description

the above code will output:
```
Traceback (most recent call last):
  File "/home/auderson/mambaforge/envs/py3.10/lib/python3.10/site-packages/IPython/core/interactiveshell.py", line 3508, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-11-6dcccd9b83bc>", line 1, in <module>
    df.insert(loc_a, "b", 0)
  File "/home/auderson/mambaforge/envs/py3.10/lib/python3.10/site-packages/pandas/core/frame.py", line 4784, in insert
    raise TypeError("loc must be int")
TypeError: loc must be int

```

### Expected Behavior

insert column correctly

### Installed Versions

<details>

/home/auderson/mambaforge/envs/py3.10/lib/python3.10/site-packages/_distutils_hack/__init__.py:33: UserWarning: Setuptools is replacing distutils.
  warnings.warn("Setuptools is replacing distutils.")

INSTALLED VERSIONS
------------------
commit           : 37ea63d540fd27274cad6585082c91b1283f963d
python           : 3.10.10.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.15.0-71-generic
Version          : #78-Ubuntu SMP Tue Apr 18 09:00:29 UTC 2023
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8
pandas           : 2.0.1
numpy            : 1.23.5
pytz             : 2023.3
dateutil         : 2.8.2
setuptools       : 67.7.2
pip              : 23.1.2
Cython           : 0.29.34
pytest           : 7.3.1
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : 1.0.3
psycopg2         : 2.9.3
jinja2           : None
IPython          : 8.13.1
pandas_datareader: None
bs4              : None
bottleneck       : None
brotli           : 
fastparquet      : None
fsspec           : None
gcsfs            : None
matplotlib       : 3.7.1
numba            : 0.57.0
numexpr          : 2.8.4
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : 11.0.0
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : 1.10.1
snappy           : None
sqlalchemy       : 1.4.46
tables           : 3.8.0
tabulate         : None
xarray           : None
xlrd             : None
zstandard        : None
tzdata           : 2023.3
qtpy             : None
pyqt5            : None

</details>

### PR DESCRIPTION
**Pull request by auderson**
- [ ] closes #53193 
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/frame.py [hunk 1]
```diff
@@ -4800,7 +4800,7 @@ def insert(
         if not allow_duplicates and column in self.columns:
             # Should this be a different kind of error??
             raise ValueError(f"cannot insert {column}, already exists")
-        if not isinstance(loc, int):
+        if not is_integer(loc):
```
Comment (mroeschke) [inline pandas/core/frame.py:4803]: Could you convert non stdlib ints to satisfy typing checks?

File: pandas/tests/frame/indexing/test_insert.py [hunk 2]
```diff
@@ -110,3 +110,8 @@ def test_insert_frame(self):
         )
         with pytest.raises(ValueError, match=msg):
             df.insert(1, "newcol", df)
+
+    def test_insert_int64_loc(self):
```
Comment (mroeschke) [inline pandas/tests/frame/indexing/test_insert.py:114]: Could you add the `GH` issue number?

**Review by mroeschke — COMMENTED** [review]
Could you add a whatsnew note in `v2.1.0.rst?`

**Comment by auderson** [comment]
Which section of the whatsnew note should I add it to, Conversion?

**Comment by mroeschke** [comment]
> Which section of the whatsnew note should I add it to, Conversion?

Sure

**Comment by auderson** [comment]
done

> > Which section of the whatsnew note should I add it to, Conversion?
> 
> Sure

**Review by topper-123 — APPROVED** [review]
LGTM.

**Comment by topper-123** [comment]
Thanks, @auderson.

**Pull request merged**